### PR TITLE
ENH: Add decryption to the cat operation

### PR DESCRIPTION
**Pull request by EmperorArthur**
I recently needed to decrypt a PDF file, and found the program would just error.  This resolves that issue.

* Ads an optional password parameter to the "cat" operation.  Allowing decryption.
* Makes decryption errors more readable for end-users.

### CODE REVIEW

**Comment by ouqilbo** [comment]
Love this. Started using pdfly a week ago and I was about to stop because I had to manipulate a password protected pdf

**Comment by Lucas-C** [comment]
Hi
Thank you for your contribution @EmperorArthur 👍
I think this PR needs to be rebased on `main`

**Comment by ouqilbo** [comment]
Hey @EmperorArthur, could you please proceed with the pr and follow the previous suggestions? As I already said, this is really beneficial when in need to use password protected pdfs.

File: pdfly/cat.py [hunk 1]
```diff
@@ -57,6 +58,7 @@ def main(
     output: Path,
     verbose: bool,
     inverted_page_selection: bool = False,
+    password: Optional[str] = None
```
Comment (MartinThoma) [inline pdfly/cat.py:61]:
```suggestion
    password: Optional[str] = None,
```

File: pdfly/cli.py [hunk 2]
```diff
@@ -95,8 +95,11 @@ def cat(
     verbose: bool = typer.Option(
         False, help="show page ranges as they are being read"
     ),
+    password: str = typer.Option(
+        None, help="Document's user or owner password."
+    ),
 ) -> None:
-    pdfly.cat.main(filename, fn_pgrgs, output, verbose)
+    pdfly.cat.main(filename, fn_pgrgs, output, verbose, password)
```
Comment (MartinThoma) [inline pdfly/cli.py:102]:
```suggestion
    pdfly.cat.main(filename, fn_pgrgs, output, verbose, password=password)
```

File: pdfly/cat.py [hunk 3]
```diff
@@ -46,9 +46,10 @@
 import sys
 import traceback
 from pathlib import Path
-from typing import List, Tuple
+from typing import List, Tuple, Optional
```
Comment (MartinThoma) [inline pdfly/cat.py:49]:
```suggestion
from typing import List, Optional, Tuple
```

File: pdfly/cat.py [hunk 4]
```diff
@@ -76,7 +78,9 @@ def main(
             if filepath not in in_fs:
                 in_fs[filepath] = open(filepath, "rb")
 
-            reader = PdfReader(in_fs[filepath])
+            reader = PdfReader(in_fs[filename])
```
Comment (MartinThoma) [inline pdfly/cat.py:81]:
```suggestion
            reader = PdfReader(in_fs[filepath])
```

**Comment by MartinThoma** [comment]
Thank you for your contribution @EmperorArthur :pray:

**Comment by MartinThoma** [comment]
@ouqilbo This will be part of the next release. You can always install the current development version like this: https://pdfly.readthedocs.io/en/latest/user/installation.html#development-version

**Pull request merged**